Formula/f/flang.rb

```diff
@@ -96,6 +94,12 @@ def install
     end
   end
 
+  def post_install
+    # Allow flang -flto to work on Linux as it expects library relative to driver.
+    # The HOMEBREW_PREFIX path is used so that `brew link` skips creating a symlink.
+    lib.install_symlink HOMEBREW_PREFIX/"lib/LLVMgold.so" if OS.linux?
+  end
```

Comment:
What happens if you do `brew unlink flang && brew link flang` after this symlink is created?

Reply:
I didn't see any issues and no conflicting symlink is attempted. This scenario is handled by https://github.com/Homebrew/brew/blob/main/Library/Homebrew/keg.rb#L706

```console
$ brew unlink flang
Unlinking /home/linuxbrew/.linuxbrew/Cellar/flang/20.1.8... 59 symlinks removed.

$ brew link flang
Linking /home/linuxbrew/.linuxbrew/Cellar/flang/20.1.8... 59 symlinks created.

$ readlink ~/.linuxbrew/lib/LLVMgold.so
../Cellar/llvm/20.1.8/lib/LLVMgold.so

$ readlink ~/.linuxbrew/opt/flang/lib/LLVMgold.so
../../../../lib/LLVMgold.so

$ readlink -f ~/.linuxbrew/opt/flang/lib/LLVMgold.so
/home/linuxbrew/.linuxbrew/Cellar/llvm/20.1.8/lib/LLVMgold.so

$ brew ruby -e 'p (Formula["flang"].opt_lib/"LLVMgold.so").resolved_path'
#<Pathname:/home/linuxbrew/lib/LLVMgold.so>
```

Reply:
Could probably also handle the resource directory and `libLTO` on macOS in the same way instead of having a `flang.cfg` too, I think?